src/main/scala/io/iohk/ethereum/consensus/ethash/difficulty/EthashDifficultyCalculator.scala

```diff
@@ -5,27 +5,37 @@ import io.iohk.ethereum.domain.BlockHeader
 import io.iohk.ethereum.utils.BlockchainConfig
 
 class EthashDifficultyCalculator(blockchainConfig: BlockchainConfig) extends DifficultyCalculator {
-  import blockchainConfig.{difficultyBombContinueBlockNumber, difficultyBombPauseBlockNumber, homesteadBlockNumber, difficultyBombRemovalBlockNumber}
+  import blockchainConfig.{
+    difficultyBombContinueBlockNumber,
+    difficultyBombPauseBlockNumber,
+    homesteadBlockNumber,
+    difficultyBombRemovalBlockNumber,
+    byzantiumBlockNumber
+  }
 
   val DifficultyBoundDivision: Int = 2048
   val FrontierTimestampDiffLimit: Int = -99
   val ExpDifficultyPeriod: Int = 100000
   val MinimumDifficulty: BigInt = 131072
+  val RelaxDifficulty: BigInt = BigInt(10).pow(6) * 3
 
   def calculateDifficulty(blockNumber: BigInt, blockTimestamp: Long, parentHeader: BlockHeader): BigInt = {
+  // calculate a fake block number for the ice-age delay https://github.com/ethereum/EIPs/blob/master/EIPS/eip-649.md
+    val fakeBlockNumber: BigInt =
+      if (blockNumber >= byzantiumBlockNumber) (blockNumber - RelaxDifficulty).max(0) else blockNumber
 
     val x: BigInt = parentHeader.difficulty / DifficultyBoundDivision
     val c: BigInt =
-      if (blockNumber < homesteadBlockNumber) {
+      if (fakeBlockNumber < homesteadBlockNumber) {
         if (blockTimestamp < parentHeader.unixTimestamp + 13) 1 else -1
       } else {
         val timestampDiff = blockTimestamp - parentHeader.unixTimestamp
         math.max(1 - timestampDiff / 10, FrontierTimestampDiffLimit)
       }
 
     val extraDifficulty: BigInt =
-      if (blockNumber < difficultyBombRemovalBlockNumber) {
-        val difficultyBombExponent = calculateBombExponent(blockNumber)
+      if (fakeBlockNumber < difficultyBombRemovalBlockNumber) {
+        val difficultyBombExponent = calculateBombExponent(fakeBlockNumber)
```

Comment:
Spec is not clear here by sayin `simply replace the use of block.number, as used in the exponential ice age component`, but i think fakeBlock number should be used only here (as paramater to `calculateBombExponent`). 
example: https://github.com/ethereum/ethereumj/pull/927/files#diff-e6100521843ae5e9fb95013c150f2129R72

Reply:
done